tasks/main.yml

```diff
@@ -47,39 +47,56 @@
+  when: nginx_app_protect_waf_state != "absent" or nginx_app_protect_dos_state != "absent"
 
-    - name: Install NGINX App Protect WAF
-      include_tasks: "{{ role_path }}/tasks/waf/install/install-app-protect-waf.yml"
-      when: nginx_app_protect_waf_enable | bool
-      tags: nginx_app_protect_install_app_protect_waf
+- name: Install NGINX App Protect WAF
+  include_tasks: "{{ role_path }}/tasks/waf/install/install-app-protect-waf.yml"
+  when: nginx_app_protect_waf_enable | bool
+  tags: nginx_app_protect_install_app_protect_waf
 
-    - name: Install NGINX App Protect DoS
-      include_tasks: "{{ role_path }}/tasks/dos/install/install-app-protect-dos.yml"
-      when: nginx_app_protect_dos_enable | bool
-      tags: nginx_app_protect_install_app_protect_dos
+- name: Install NGINX App Protect DoS
+  include_tasks: "{{ role_path }}/tasks/dos/install/install-app-protect-dos.yml"
+  when: nginx_app_protect_dos_enable | bool
+  tags: nginx_app_protect_install_app_protect_dos
 
-    - name: Remove NGINX App Protect license
-      include_tasks: "{{ role_path }}/tasks/common/install/remove-license.yml"
-      when: nginx_app_protect_remove_license | bool
-      tags: nginx_app_protect_remove_license
+- name: Remove NGINX App Protect license
+  include_tasks: "{{ role_path }}/tasks/common/install/remove-license.yml"
+  when: nginx_app_protect_remove_license | bool
+  tags: nginx_app_protect_remove_license
 
-    - name: Configure NGINX App Protect
-      include_tasks: "{{ role_path }}/tasks/waf/config/configure-app-protect-waf.yml"
-      when: nginx_app_protect_configure | bool
-      tags: nginx_app_protect_configure
-  when: nginx_app_protect_waf_state != "absent"
+- name: Configure NGINX App Protect
+  include_tasks: "{{ role_path }}/tasks/waf/config/configure-app-protect-waf.yml"
+  when: nginx_app_protect_configure | bool
+  tags: nginx_app_protect_configure
 
 - name: Ensure NGINX App Protect is running
   meta: flush_handlers
 
-- name: Remove NGINX App Protect
+- name: Remove NGINX App Protect WAF
   block:
-    - name: Remove NGINX App Protect package
+    - name: Remove NGINX App Protect license
+      include_tasks: "{{ role_path }}/tasks/common/install/remove-license.yml"
+      when: nginx_app_protect_remove_license | bool
+      tags: nginx_app_protect_remove_license
+
+    - name: Remove NGINX App Protect WAF package
       include_tasks: "{{ role_path }}/tasks/waf/install/install-app-protect-waf.yml"
-      tags: nginx_app_protect_remove
+      tags: nginx_app_protect_waf_remove
 
     - name: Disable NGINX App Protect config
       include_tasks: "{{ role_path }}/tasks/waf/config/configure-app-protect-waf.yml"
       when: nginx_app_protect_configure | bool
-      tags: nginx_app_protect_remove_config
+      tags: nginx_app_protect_waf_remove_config
   when: nginx_app_protect_waf_state == "absent"
+
+- name: Remove NGINX App Protect WAF
+  block:
+    - name: Remove NGINX App Protect license
```

Comment:
This step is shared between both WAF/DoS. We could maybe put it outside these blocks.